vmm/sandbox/src/qemu/mod.rs

```diff
@@ -70,21 +71,15 @@ pub(crate) const QEMU_START_TIMEOUT_IN_SEC: u64 = 10;
 // but skip all the fields serialization.
 #[derive(Default, Serialize, Deserialize)]
 pub struct QemuVM {
-    #[serde(skip)]
     id: String,
-    #[serde(skip)]
     config: QemuConfig,
     #[serde(skip)]
     devices: Vec<Box<dyn QemuDevice + Sync + Send>>,
     #[serde(skip)]
     hot_attached_devices: Vec<Box<dyn QemuHotAttachable + Sync + Send>>,
```

Comment:
the hot_attached_devices should be recovered when restart, otherwise the devices of the container will be residual after the container is restarted.